plugins/plugin-svn/che-plugin-svn-ext-ide/src/main/java/org/eclipse/che/plugin/svn/ide/merge/MergePresenter.java

```diff
@@ -99,62 +106,84 @@ public void merge() {
-                                                       if (!"file".equals(item.getNodeKind())) {
-                                                           children.add(new SubversionItemNode(item));
-                                                       }
-                                                   }
-                                               }
-
-                                               subversionTreeNode.setChildren(children);
-                                               view.setRootNode(subversionTreeNode);
-                                               view.show();
-                                               validateSourceURL();
-                                           }
-                                       })
-                       .catchError(new Operation<PromiseError>() {
-                                           @Override
-                                           public void apply(PromiseError error) throws OperationException {
-                                               notificationManager.notify(error.getMessage(), FAIL, FLOAT_MODE);
-                                           }
-                                       });
-            }
-        }).catchError(new Operation<PromiseError>() {
-            @Override
-            public void apply(PromiseError error) throws OperationException {
-                notificationManager.notify(error.getMessage(), FAIL, FLOAT_MODE);
-            }
-        });
+        merge(project, resources, null, null);
+    }
+
+    private void merge(final Project project, final Resource[] resources, String userName, final String password) {
+        service.info(project.getLocation(), toRelative(project, resources[0]), "HEAD", false, userName, password)
+               .then(new Operation<InfoResponse>() {
+                   @Override
+                   public void apply(InfoResponse response) throws OperationException {
+                       if (response.getErrorOutput() != null && !response.getErrorOutput().isEmpty()) {
+                           printErrors(response.getErrorOutput(), constants.commandInfo());
+                           notificationManager.notify("Unable to execute subversion command", FAIL, FLOAT_MODE);
+                           return;
+                       }
+
+                       mergeTarget = response.getItems().get(0);
+                       view.targetTextBox().setValue(mergeTarget.getRelativeURL());
+
+                       String repositoryRoot = mergeTarget.getRepositoryRoot();
+
+                       service.info(project.getLocation(), repositoryRoot, "HEAD", true)
+                              .then(new Operation<InfoResponse>() {
+                                  @Override
+                                  public void apply(InfoResponse response) throws OperationException {
+                                      if (response.getErrorOutput() != null && !response.getErrorOutput().isEmpty()) {
```

Comment:
isNullOrEmpty can't be used here because getErrorOutput returns list of srting. BTW DTO should return empty collection instead of null